var/spack/repos/builtin/packages/snakemake/package.py

```diff
@@ -96,12 +101,13 @@ class Snakemake(PythonPackage):
     )
     depends_on("py-snakemake-interface-storage-plugins@3", type=("build", "run"), when="@8:")
 
+    depends_on("py-snakemake-interface-report-plugins@1.1:1", type=("build", "run"), when="@8.22:")
     depends_on("py-snakemake-interface-report-plugins@1", type=("build", "run"), when="@8.5:")
-    depends_on("py-stopit", type=("build", "run"))
+    depends_on("py-stopit", type=("build", "run"), when="@:8.19.0")
     depends_on("py-tabulate", type=("build", "run"))
     depends_on("py-throttler", type=("build", "run"), when="@7:")
-    depends_on("py-toposort@1.10:1", type=("build", "run"), when="@8.4.12:")
-    depends_on("py-toposort@1.10:", type=("build", "run"), when="@7.24.0:")
+    depends_on("py-toposort@1.10:1", type=("build", "run"), when="@8.4.12:8.21")
+    depends_on("py-toposort@1.10:", type=("build", "run"), when="@7.24.0:8.21")
```

Comment:
This makes it a legacy dependency, we usually push these to the end of package definition.